pynsot/commands/cmd_protocol_types.py

```diff
@@ -0,0 +1,300 @@
+    """
+    List existing Protocol Types for a Site.
+
+    You must provide a Site ID using the -s/--site-id option.
+
+    When listing Protocol Types, all objects are displayed by default. You
+    optionally may lookup a single Protocol Types by ID using the -i/--id option.
+    The ID can either be the numeric ID of the Protocol Type.
+    """
+    data = ctx.params
+
+    # If we provide ID, show more fields
+    if any([id, name]):
+        display_fields = VERBOSE_FIELDS
+    else:
+        display_fields = DISPLAY_FIELDS
+
+    # If we aren't passing a sub-command, just call list(), otherwise let it
+    # fallback to default behavior.
+    if ctx.invoked_subcommand is None:
+        ctx.obj.list(
+            data, display_fields=display_fields,
+            verbose_fields=VERBOSE_FIELDS
+        )
+
+@list.command()
+@click.pass_context
+def protocols(ctx, *args, **kwargs):
+    """Recursively get all protocols of an Protocol Type."""
+    callbacks.list_subcommand(
+        ctx, display_fields=PROTOCOL_DISPLAY_FIELDS, my_name=ctx.info_name
+    )
+
+
+# Remove
+@cli.command()
+@click.option(
+    '-i',
+    '--id',
+    metavar='ID',
+    type=types.NATURAL_KEY,
+    help='Unique ID of the Protocol Type being deleted.',
+    required=True,
+)
+@click.option(
+    '-s',
+    '--site-id',
+    metavar='SITE_ID',
+    type=int,
+    help='Unique ID of the Site this Protocol Type is under.',
+    callback=callbacks.process_site_id,
+    required=True,
+)
+@click.pass_context
+def remove(ctx, id, site_id):
+    """
+    Remove an Protocol Type.
+
+    You must provide a Site ID using the -s/--site-id option.
+
+    When removing an Protocol Type, you must provide the ID of the Protocol Type using
+    -i/--id.
+
+    You may retrieve the ID for an Protocol Type by parsing it from the list of
+    Protocol Types for a given Site:
+
+        nsot protocol_types list --site-id <site_id> | grep <protocol_type name>
+    """
+    data = ctx.params
+    ctx.obj.remove(**data)
+
+
+# Update
+@cli.command()
+@click.option(
+    '-r',
+    '--required_attributes',
+    metavar='ATTRIBUTE',
+    type=str,
+    help='The name of a Protocol attribute. This option can be provided multiple times, once per attribute.',
+    multiple=True,
+    callback=callbacks.transform_attributes,
+)
+@click.option(
+    '-d',
+    '--description',
+    metavar='DESCRIPTION',
+    type=str,
+    help='The description for this Protocol Type.',
+)
+@click.option(
+    '-i',
+    '--id',
+    metavar='ID',
+    type=types.NATURAL_KEY,
+    help='Unique ID of the Protocol Type being updated.',
+    required=True,
+)
+@click.option(
+    '-n',
+    '--name',
+    metavar='NAME',
+    type=str,
+    help='The name of the Protocol Type.',
+)
+@click.option(
+    '-s',
+    '--site-id',
+    metavar='SITE_ID',
+    type=int,
+    help='Unique ID of the Site this Protocol Type is under.',
+    callback=callbacks.process_site_id,
+    required=True,
```

Comment:
We can't have `required=True` on `-s/--site-id` because we want the `default_site` from `.pynsorc` to be used if it exists. That's the purpose of the `process_site_id` callback, which will display the required argument error if the `default_site` value cannot be read.